### PR TITLE
[Linux/ARM32] objcopy: Support cross-build  for stripping symbols.

### PR DESCRIPTION
**Pull request by leemgs**
Currently, the 'objcopy' does only support native-build since the commit number 4efef62f0ead16b4f8ca2c6ccf0a77ce112f73b9 Let's add cross-build as well as native-build to strip symbols.

Signed-off-by: Geunsik Lim geunsik.lim@samsung.com
Signed-off-by: MyungJoo Ham myungjoo.ham@samsung.com
Signed-off-by: Prajwal A N an.prajwal@samsung.com

/CC: @myungjoo , @prajwal-aithal

### CODE REVIEW

**Comment by leemgs** [comment]
This PR can solve Issue # https://github.com/dotnet/coreclr/issues/3945

**Comment by leemgs** [comment]
\CC: @jkotas , @manu-silicon , @benpye

**Comment by benpye** [comment]
Looks like arm64 would still be broken, we should probably add a variable to the toolchain file with the build triple (and reduce duplication there) which could then be used here to get the right objcopy, what do you think?

**Comment by leemgs** [comment]
@benpye ,  :100:  Agree. Please, refer to https://github.com/dotnet/coreclr/pull/3946  that we submitted to reduce duplication in advance.  But, Le't do next activities step-by-step to avoid regression and to avoid build-break of other command.

**Comment by benpye** [comment]
Ah then sure, if that's also identified as an issue then yes this is a step toward. LGTM.

File: CMakeLists.txt [hunk 1]
```diff
@@ -166,15 +166,19 @@ else()
       if (STRIP STREQUAL "STRIP-NOTFOUND")
           message(FATAL_ERROR "strip not found")
       endif()
+    endif (CMAKE_SYSTEM_NAME STREQUAL Darwin)
 
-    else (CMAKE_SYSTEM_NAME STREQUAL Darwin)
-
+    if (CMAKE_SYSTEM_NAME STREQUAL Linux)
```
Comment (benpye) [inline CMakeLists.txt]: This should probably be an elseif.

**Comment by leemgs** [comment]
> This should probably be an elseif.

@benpye , Thanks. updated.

**Comment by leemgs** [comment]
Here is  check result of the "stripped" attribute after updating. 

invain@u1404:/work/dotnet/arm-32/coreclr/bin/Product/Linux.arm.Release$ file ./libcoreclr.so
./libcoreclr.so: ELF 32-bit LSB  shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, BuildID[sha1]=7fb833b5a10fe53696d286e1fe783f4d3f1f3fac, **stripped**
invain@u1404:/work/dotnet/arm-32/coreclr/bin/Product/Linux.arm.Release$ 

invain@u1404:/work/dotnet/arm-32/coreclr/bin/Product/Linux.arm.Release$ file ./mscorlib.dll 
./mscorlib.dll: PE32 executable (DLL) (console) **ARMv7** Thumb Mono/.Net assembly, for MS Windows

**Comment by jkotas** [comment]
CentOS and Ubuntu builds fail with:

```
CMake Error at CMakeLists.txt:171 (if):
    if given arguments:
    "EQUAL" "1"
    Unknown arguments specified
```

**Comment by leemgs** [comment]
> CentOS and Ubuntu builds fail with:

@jkotas , Thanks, Submitted PR again.

**Comment by jkotas** [comment]
@dotnet-bot test this please

File: CMakeLists.txt [hunk 2]
```diff
@@ -166,15 +166,17 @@ else()
...
+      if(DEFINED ENV{CROSSCOMPILE})
+        find_program(OBJCOPY arm-linux-gnueabihf-objcopy)
+      else()
+        find_program(OBJCOPY objcopy)
+      endif()
       if (OBJCOPY STREQUAL "OBJCOPY-NOTFOUND")
           message(FATAL_ERROR "objcopy not found")
       endif()
-    endif (CMAKE_SYSTEM_NAME STREQUAL Darwin)
+    endif ()
```
Comment (manu-silicon) [inline CMakeLists.txt]: I think it should still handle Mac OS X which is lost here by the removal of the if condition. Also it should handle both cross compilation for arm and arm64.

Comment (leemgs) [inline CMakeLists.txt]: @manu-silicon , :+1:  , I have appended the `if` statement for handling arm64 even though I don't currently follow up the arm64 issue.

**Comment by jkotas** [comment]
@dotnet-bot test this please

**Comment by leemgs** [comment]
It's strange.  It seems that CI system has broken for 3days. Actually, many PRs have now been pending for a long time. Coul you share us a person that is in charge of the normal operation of CI? 

We can see the many PRs (e.g., red symbols) at https://github.com/dotnet/coreclr/pulls

**Comment by jkotas** [comment]
@leemgs We have several people working on getting CI back on track. We are running into a lot of stability issues with Jenkins...

**Comment by jkotas** [comment]
@dotnet-bot test this please

**Comment by jkotas** [comment]
@mikem8361 Could you please take a look at this change?

**Comment by mikem8361** [comment]
LGTM

**Comment by jkotas** [comment]
@dotnet-bot test OSX x64 Checked Build and Test   please

**Comment by jkotas** [comment]
@dotnet-bot test Ubuntu x64 Checked Build and Test  please

**Comment by jkotas** [comment]
@dotnet-bot test this please

**Pull request merged**